delphi-checks/src/main/java/au/com/integradev/delphi/checks/CaseStatementSizeCheck.java

```diff
@@ -28,13 +30,40 @@
 @Rule(key = "CaseStatementSize")
 public class CaseStatementSizeCheck extends DelphiCheck {
   private static final String MESSAGE = "Replace this 'case' statement with an 'if' statement.";
+  private static final int REQUIRED_CASE_ITEMS = 2;
 
   @Override
   public DelphiCheckContext visit(CaseStatementNode caseStatement, DelphiCheckContext context) {
-    if (caseStatement.getCaseItems().size() < 2) {
+    if (countCaseItems(caseStatement) < REQUIRED_CASE_ITEMS) {
       reportIssue(context, caseStatement.getChild(0), MESSAGE);
     }
 
     return super.visit(caseStatement, context);
   }
+
+  private static int countCaseItems(CaseStatementNode caseStatement) {
+    return caseStatement.getCaseItems().stream()
+        .mapToInt(CaseStatementSizeCheck::countCaseItem)
+        .sum();
+  }
+
+  private static int countCaseItem(CaseItemStatementNode caseItem) {
+    return caseItem.getExpressions().stream()
+        .mapToInt(
+            expression -> {
+              if (expression instanceof RangeExpressionNode) {
+                // HACK:
+                //  We're assuming that the range expression actually represents 2 or more values,
+                //  and isn't something silly like 0..0.
+                //  Constant expression evaluation would be needed to get the actual number of
+                //  values in the range.
+                //
+                //  See:
+                //   https://github.com/integrated-application-development/sonar-delphi/issues/116
```

Comment:
I don't think this is a hack -  it seems sensible to exclude case statements with a range expression, since they're not directly mappable onto an if statement.

I don't think the mindset should be that we're hacking the behaviour of this rule until we have constant expression evaluation - I would argue that would be overkill in this case. We also probably want to keep the hack comments isolated to where they're most hacky.

Reply:
I'm of two minds about it because it would be nice to be able to statically recognize stupid patterns like a range of 1.
But on balance, I think you're right - it's overkill to consider this a hack.

Reply:
I've changed the comment.